File: js/example-post.js
const hmac = require("./hmac");
const fetch = require("node-fetch");

// An example post request to find nearby sites from the Sites API
async function examplePost(secretKey, sharedKey, nepOrganization) {
  let date = new Date();

  let options = {
    date,
    secretKey,
    sharedKey,
    nepOrganization,
    requestURL:
      "https://gateway-staging.ncrcloud.com/site/sites/find-by-criteria?pageNumber=0&pageSize=200",
    httpMethod: "POST",
    contentType: "application/json",
  };

  const hmacAccessKey = hmac(options);

  let body = {
    sort: [
      {
        column: "siteName",
        direction: "asc",
      },
    ],
  };

  var requestOptions = {
    method: options.httpMethod,
    headers: {
      "Content-Type": options.contentType,
      Authorization: `AccessKey ${hmacAccessKey}`,
      "nep-organization": options.nepOrganization,
      Date: date.toGMTString(),
    },
    body: JSON.stringify(body),
  };

  const response = await fetch(options.requestURL, requestOptions);

  const status = response.status;

  // PUT requests return 204 with no response.
  if (status == 204) {
    console.log("Success");
  } else {
    let data = await response.json();
    return {
      status,
      data,
    };
  }
}

module.exports = examplePost;

File: python/examplePost.py
from hmacHelper import hmacHelper
from datetime import datetime
from datetime import timezone
import requests
import json


def examplePost(secretKey="9a27e38da2194950a04fcba431ad9b99", sharedKey="139aa99b486c4848821929b85b0b5b03", nepOrganization="test-drive-ed002b0e2b234b32858eb"):
    
    now = datetime.now(tz=timezone.utc)
    now = datetime(now.year, now.month, now.day, hour=now.hour,
                   minute=now.minute, second=now.second)

    requestURL = "https://api.ncr.com/security/authentication/login"
    httpMethod = 'POST'
    contentType = 'application/json'

    hmacAccessKey = hmacHelper(sharedKey, secretKey, now, httpMethod,
                               requestURL, contentType, None, None, None, nepOrganization, None)

    utcDate = now.strftime('%a, %d %b %Y %H:%M:%S GMT')
    headers = {
        "Date": utcDate,
        "Content-Type": contentType,
        "Authorization": "AccessKey " + hmacAccessKey,
        "nep-organization": nepOrganization
    }

    data = {
    }

    payload = json.dumps(data)

    request = requests.post(requestURL, data=payload, headers=headers)

    res = dict()
    res['status'] = request.status_code
    res['data'] = request.json()

    json_formatted = json.dumps(res, indent=2)
    print(json_formatted)

    return res


examplePost()
